feat: add google sheet and csv support (#129)
empirical-run/empirical: docs/dataset/basics.mdx

Specify a path to the JSONL file. Each line of the file should be a valid JSON object.
On import, the keys of this JSON will be converted into inputs of the sample.

If using relative paths, the path is treated relative to the configuration file.

```json
"dataset": {
  "path": "HumanEval.jsonl"
}
```

## Import from CSV
Specify a path to the CSV file in the `empiricalrc.json`. If using relative paths, the path is treated relative to the configuration file.

```json
"dataset": {
  "path": "foo.csv"
}
```

The CSV file should contain headers.
The lines of the file are converted into dataset inputs with column header names as the name of the parameter. For example:
```csv foo.csv
name,age
John,25
```
The above CSV gets converted into the following dataset object: 
```json
"dataset": {
  "samples": [
    {
      "inputs": {
        "name": "John",
        "age": "25"
      }
    }
  ]
}
```

> The above conversion enables you to create prompt with placeholders. For example: 

Reviewer: ```suggestion
The above conversion enables you to create a prompt with placeholders. For example: 
```

```json
{
  "prompt": "Your name is {{name}} and you are a helpful assistant..."
}
```

## Import from Google Sheets

> Specify a path to the Google sheet in the `empiricalrc.json`. 

Reviewer: ```suggestion
Specify a path to the Google sheet in the `empiricalrc.json` file.
```

```json empiricalrc.json
"dataset": {
  "path": "https://docs.google.com/spreadsheets/d/1AsMekKCG74m1PbBZQN_sEJgaW0b9Xarg4ms4mhG3i5k"
}
```

> Ref: [chatbot example](https://github.com/empirical-run/empirical/tree/main/examples/chatbot)

Reviewer: ```suggestion
Refer to our [chatbot example](https://github.com/empirical-run/empirical/tree/main/examples/chatbot) which uses this dataset.
```
